button-merchant/src/test/java/com/usebutton/merchant/ButtonApiImplTest.java

```diff
@@ -151,4 +161,107 @@ public void postUserActivity_validateRequest() throws Exception {
         assertEquals("AUG", requestBody.getString("currency"));
         assertEquals("merchant-library", requestBody.getString("source"));
     }
+
+    @Test
+    public void postOrder_validateRequest() throws Exception {
```

Comment:
What do you think about breaking this down into smaller tests for different elements in the payload?